lib/puppet/parser/functions.rb

```diff
@@ -44,15 +44,10 @@ def self.autoloader
   # @api private
   class AnonymousModuleAdapter < Puppet::Pops::Adaptable::Adapter
     attr_accessor :module
-  end
 
-  # Get the module that functions are mixed into corresponding to an
-  # environment
-  #
-  # @api private
-  def self.environment_module(env)
-    AnonymousModuleAdapter.adapt(env) do |a|
-      a.module ||= Module.new do
+    def self.create_adapter(o)
+      a = super(o)
```

Comment:
Sure, I just left the existing local variable name, but I can call it `adapter` which is what I think the `a` was short for.

Reply:
the `o` was even more confusing - I could deduce that `a` was for `adapter`, but I have no idea what `o` is supposed to be.

Reply:
ahh, that was copied from other `create_adapter` calls. I think it just stands for "object" because in general adapters can adapt any Ruby object. However I could specify it more here since this adapter is specifically for environments.

Reply:
I'll rename it `e` in that case, right?  :trollface:

Reply:
are there more `create_adapter` calls than in this PR? I'm wondering if we should change them all. Even just `object` is clearer to me.

Reply:
yes, and I can do that.